### ISSUE TITLE
Versions are for endpoints

### ISSUE THREAD

**Issue by martinthomson** [quicwg/base-drafts#3808]
The invariants spec currently defines Version thusly:

> QUIC versions are identified with a 32-bit integer, encoded in network byte order.

This implies that packets in which this field is present are packets for connections that use the associated version.  An overzealous, but not unreasonable interpretation of this might lead to a middlebox looking at Version=N and assuming that the packet conforms to QUIC version N.  Actions taken on that basis might be detrimental, especially if we consider options like those proposed in draft-duke-quic-version-aliasing or draft-kazuho-quic-authenticated-handshake.

In other words, this commits us to something we already know we might not want to commit to.

It would be better to simply state the format of the field and that it is used by endpoints to identify the version.  I have done this in #3785.

**Comment by larseggert** [comment]
There seems to be support for this issue and #3785, and the change is editorial.

**Comment by larseggert** [comment]
As an individual, I wonder if it would be worthwhile to add text to the draft that explicitly calls out that it is not safe for anybody other than the endpoints to assume knowledge about what protocol version an observed version number refers to. I think this it the implicit goal, making it explicit is IMO more direct.

(This could also be something for the ops-drafts to talk about though.)

**Comment by LPardue** [comment]
> There seems to be support for this issue and #3785, and the change is editorial.

@martinthomson relabelled the PR as design a few days ago, so it this potentially a design issue?

**Comment by larseggert** [comment]
Looking at the diff, I don't really see a need for labelling it design. @martinthomson, why the design label?

**Comment by MikeBishop** [comment]
It's a subtle change, but it is a change in the design of the invariants.  It's saying that a packet marked version 0x13 isn't necessarily carrying a packet encoded in version 0x13, even if an observer speaks that version.  Previously, we made the assertion that the version field indicated what version was used in the packet.

**Comment by LPardue** [comment]
Let's run this through the design issue process to be sure.

**Comment by larseggert** [comment]
So it seems we have a resolution in #3785. Labelling as such.
